base/src/AudioToTextXForm.cpp

```diff
@@ -0,0 +1,212 @@
+	~Detail() {}
+
+	whisper_full_params fetchDefaultParams() {
+		auto samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_GREEDY;
+		switch (mProps.samplingStrategy)
+		{
+		case AudioToTextXFormProps::DecoderSamplingStrategy::GREEDY:
+			samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_GREEDY;
+			break;
+		case AudioToTextXFormProps::DecoderSamplingStrategy::BEAM_SEARCH:
+			samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_BEAM_SEARCH;
+			break;
+		default:
+			throw AIPException(AIP_FATAL, "Unknown Sampling Strategy");
+		}
+		return whisper_full_default_params(samplingStrategy);
+	}
+
+	void setProps(AudioToTextXFormProps& props)
+	{
+		mProps = props;
+	}
+
+public:
+	framemetadata_sp mOutputMetadata;
+	std::string mOutputPinId;
+	std::vector<float> inputAudioBuffer;
+	AudioToTextXFormProps mProps;
+	int mFrameType;
+	whisper_context *mWhisperContext = NULL;
+	whisper_full_params mWhisperFullParams;
+	whisper_context_params mWhisperContextParams;
+};
+
+AudioToTextXForm::AudioToTextXForm(AudioToTextXFormProps _props) : Module(TRANSFORM, "AudioToTextXForm", _props)
+{
+	mDetail.reset(new Detail(_props));
+}
+
+AudioToTextXForm::~AudioToTextXForm() {}
+
+bool AudioToTextXForm::validateInputPins()
+{
+	//TODO: Reject any audio pin that has a samplingRate!=16KHz 
+	//https://github.com/Apra-Labs/ApraPipes/issues/325
+	if (getNumberOfInputPins() != 1)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateInputPins size is expected to be 1. Actual<" << getNumberOfInputPins() << ">";
+		return false;
+	}
+
+	framemetadata_sp metadata = getFirstInputMetadata();
+
+	FrameMetadata::FrameType frameType = metadata->getFrameType();
+
+	if (frameType != FrameMetadata::AUDIO)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateInputPins input frameType is expected to be Audio. Actual<" << frameType << ">";
+		return false;
+	}
+
+	return true;
+}
+
+bool AudioToTextXForm::validateOutputPins()
+{
+	if (getNumberOfOutputPins() != 1)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateOutputPins size is expected to be 1. Actual<" << getNumberOfOutputPins() << ">";
+		return false;
+	}
+
+	framemetadata_sp metadata = getFirstOutputMetadata();
+	FrameMetadata::FrameType frameType = metadata->getFrameType();
+	if (frameType != FrameMetadata::TEXT)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateOutputPins input frameType is expected to be TEXT. Actual<" << frameType << ">";
+		return false;
+	}
+
+	return true;
+}
+
+void AudioToTextXForm::addInputPin(framemetadata_sp& metadata, string& pinId)
+{
+	Module::addInputPin(metadata, pinId);
+	mDetail->mOutputMetadata = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::TEXT));
+	mDetail->mOutputMetadata->copyHint(*metadata.get());
+	mDetail->mOutputPinId = addOutputPin(mDetail->mOutputMetadata);
+}
+
+bool AudioToTextXForm::init()
+{
+	mDetail->inputAudioBuffer.reserve(mDetail->mProps.bufferSize + 16000); //16000 is 1 second worth of samples since data is captured at 16KHz
```

Comment:
Instead of hardcoding 16000, We can define a constant Variable for the same

Reply:
done